packages/core/test/lib/transports/base.test.ts

```diff
@@ -0,0 +1,344 @@
+    await transport.flush(1000);
+    expect(mockBuffer.drain).toHaveBeenCalledTimes(1);
+    expect(mockBuffer.drain).toHaveBeenLastCalledWith(1000);
+    /* eslint-enable @typescript-eslint/unbound-method */
+  });
+
+  describe('send', () => {
+    it('constructs a request to send to Sentry', async () => {
+      const transport = createTransport({}, req => {
+        expect(req.category).toEqual(ERROR_TRANSPORT_CATEGORY);
+        expect(req.body).toEqual(serializeEnvelope(ERROR_ENVELOPE));
+        return resolvedSyncPromise({ statusCode: 200 });
+      });
+      await transport.send(ERROR_ENVELOPE, ERROR_TRANSPORT_CATEGORY);
+    });
+
+    describe('Rate-limiting', () => {
+      it('back-off using Retry-After header', async () => {
+        const retryAfterSeconds = 10;
+        const beforeLimit = Date.now();
+        const withinLimit = beforeLimit + (retryAfterSeconds / 2) * 1000;
+        const afterLimit = beforeLimit + retryAfterSeconds * 1000;
+
+        jest
+          .spyOn(Date, 'now')
+          // 1st event - isRateLimited - FALSE
+          .mockImplementationOnce(() => beforeLimit)
+          // 1st event - updateRateLimits
+          .mockImplementationOnce(() => beforeLimit)
+          // 2nd event - isRateLimited - TRUE
+          .mockImplementationOnce(() => withinLimit)
+          // 3rd event - isRateLimited - FALSE
+          .mockImplementationOnce(() => afterLimit)
+          // 3rd event - updateRateLimits
+          .mockImplementationOnce(() => afterLimit);
+
+        let transportResponse: TransportMakeRequestResponse = {
+          headers: {
+            'retry-after': `${retryAfterSeconds}`,
+          },
+          statusCode: 429,
+        };
+        const transport = createTransport({}, _ => {
+          return resolvedSyncPromise(transportResponse);
+        });
+
+        try {
+          await transport.send(ERROR_ENVELOPE, ERROR_TRANSPORT_CATEGORY);
+        } catch (res) {
+          expect(res.status).toBe('rate_limit');
+          expect(res.reason).toBe(`Too many error requests, backing off until: ${new Date(afterLimit)}`);
+        }
+
+        // clear rate-limit headers and statusCode
+        transportResponse = {
+          statusCode: 200,
+        };
+        try {
+          await transport.send(ERROR_ENVELOPE, ERROR_TRANSPORT_CATEGORY);
+        } catch (res) {
+          expect(res.status).toBe('rate_limit');
+          expect(res.reason).toBe(`Too many error requests, backing off until: ${new Date(afterLimit)}`);
+        }
+
+        const res = await transport.send(ERROR_ENVELOPE, ERROR_TRANSPORT_CATEGORY);
+        expect(res.status).toBe('success');
+      });
+
+      it('back-off using X-Sentry-Rate-Limits with single category', async () => {
+        const retryAfterSeconds = 10;
+        const beforeLimit = Date.now();
+        const withinLimit = beforeLimit + (retryAfterSeconds / 2) * 1000;
+        const afterLimit = beforeLimit + retryAfterSeconds * 1000;
```

Comment:
Just a suggestion: Since these variables are identical in 5 tests, I think it would make sense to extract them into a helper function.

Reply:
Extracted!